resources/versioning.md

```diff
@@ -0,0 +1,92 @@
+
+**In other terms, if the users MAY have to do more steps than just downloading the new Meilisearch package and running it, a new MAJOR is needed.**
+
+Examples of changes making the code break and then, involving increasing the MAJOR:
+
+- Any change related to the public API the SDK exposes like:
+    - Renaming a method name or classes
+    - Removing or changing a positional argument of a method.
+
+## Reasons why we should release a major version
+
+### New runtimes adoption
+
+We don't have a strict policy yet regarding the minimum support of the language version. Example:
+
+Ruby SDK has a `required_ruby_version >= 2.6`, which prevents users from using the gem if they run their apps with a `Ruby 2.5`.
+
+It is often good to only allow newer versions of the runtimes because they are maintained :sparkles: and there are always new features we can adopt in the code.
+
+But there is a price for that. When we make that cut, we reduce the number of users who can adopt our SDK/service.
+
+Currently, all supported versions are set in the CI of each repository.
+
+This is the list of minimum supported language version runtimes*:
+
+| Language/Framework | Version |
+| ------------------ | ------- |
+| .NET | netstandard2.0 |
+| Dart | >= 2.12.0 |
+| Go | >= 1.16 |
+| Java | >= 8.0 |
+| JavaScript/Node | >= 14 |
+| Swift | >= 5.2 |
+| PHP | >= 7.4 |
+| Python | >= 3.7 |
+| Ruby | >= 2.6 |
+| Rust | N/A |
+
+When an SDK drops a minimum required version of the runtime, we release a new MAJOR version of the SDK.
+
+\* This list may need to be updated. Always check the current version in the integration's repository.
+
+## Correlation between Meilisearch engine versioning and the SDKs
+
+SDKs do not follow the Meilisearch engine version. It means that there is no correlation between SDK's version and Meilisearch; an SDK may be in version `v10.0.0` while the Meilisearch engine is in `v1.0.0`.
+
+Suppose the version of an SDK is the same as the latest version of Meilisearch. It is just a happy coincidence.
+
+A major release in Meilisearch may also not lead to a major release in the SDKs. For example, if we get a Meilisearch v2 in the future, there is no guarantee that the Meilisearch Dart SDK will update from v10 to v11.
+
+You can find out which version of the Meilisearch engine an SDK is compatible with by going to the `Compatibility with Meilisearch` section of every SDK README page. For example, the [README](https://github.com/meilisearch/meilisearch-ruby/#-compatibility-with-meilisearch) page of the ruby SDK.
+
+To illustrate, the idea is to have a similar table in every repository.
+
+| Meilisearch engine version | Integration version |
+| -------------------------- | ------------------- |
+| v0.30 | v0.14.0 |
+| v1.0 | >= v0.15.0 |
+| v1.1 | v0.18.0 |
+
+Given the example above, it means that the integration in v0.15 works with Meilisearch v1.1, but may only some of the new features introduced in v1.1 will work with this integration version.
```

Comment:
I changed from:

> Given the example above, it means that the integration in v0.15 works with Meilisearch v1.1, but may only some of the new features introduced in v1.1 will work with this integration version.

to: 

> Given the table above, a user using Meilisearch v1.1 does not need to have the v0.18 version of its SDK.
Of course, some features will not be present, but they should still work.

WDYT?

Reply:
Are you sure you need to talk about this? Starting from Meilisearch v1.0, every SDK will be compatible with any version of Meilisearch >= v1.0
I'm not sure the users would understand that "it is compatible, but some features would be missing". From my point of view, if some features are missing, it's not really compatible

Reply:
It's part of the confusion I put [here](https://github.com/meilisearch/integration-guides/issues/237#issuecomment-1422986421)
You need to define what is "compatible" for you

Reply:
I left a comment [here](https://github.com/meilisearch/integration-guides/issues/237#issuecomment-1431882992) where I expect we will agree with the term.
 
So, since the versioning table is not helpful, I'll remove it from this guide. And in the future, we can work on a better version where we can generate it using the code samples.